pytools/graph.py

```diff
@@ -410,6 +411,62 @@ def compute_induced_subgraph(graph: Mapping[NodeT, Set[NodeT]],
 # }}}
 
 
+# {{{ get_graph_dot_code
+
+def get_graph_dot_code(graph: Mapping[T, Collection[T]],
+                       edge_labels: Optional[Mapping[Tuple[T, T], str]] = None) \
+                       -> str:
+    """
+    Create a visualization of the graph *graph* in the
+    `dot <http://graphviz.org/>`__ language.
+
+    :arg graph: A :class:`collections.abc.Mapping` representing a directed
+        graph. The dictionary contains one key representing each node in the
+        graph, and this key maps to a :class:`collections.abc.Collection` of nodes
+        that are connected to the node by outgoing edges.
+
+    :arg edge_labels: An optional :class:`collections.abc.Mapping` of edge labels
+        between pairs of nodes.
+
+    :returns: A string in the `dot <http://graphviz.org/>`__ language.
+    """
+    from pytools import UniqueNameGenerator
+    id_gen = UniqueNameGenerator(forced_prefix="mynode")
+
+    from pytools.graphviz import dot_escape
+
+    if edge_labels is None:
+        edge_labels = {}
+
+    node_to_id = {}
+    edges = []
```

Comment:
I think this works without explicitly building `edges`.

Reply:
Fixed in d8c6ac9